### PR TITLE
Reviews

### PR DESCRIPTION
**Pull request by Pallavi25Kishore**
@mictro97 @shaoyuchen3213 

Form.jsx is not yet fully complete - need to add more features  - but feel free to review uptil the code that has already been added in form.jsx - the stuff added so far is complete in its own right 

The PR contains code for the following- 
1. Implementing recommendations feature
2. Implement username display 
3. Implement seller response display
4. Helpfullness feature
5. Sort options feature 
6. Rating Summary feature 
7. Rating Bars feature
8. Implement star rating filter
9. Rating breakdown - remove all filters feature
10. Implement factors breakdown feature 
11. tests for all the above 
12. started work on form.jsx

### CODE REVIEW

File: client/src/components/Reviews/RatingBars.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const numOfThreeStars = Number(ratings['3']) || 0;
+  const Three = (numOfThreeStars / totalNumberOfRatings) * 100;
+
+
+  const numOfTwoStars = Number(ratings['2']) || 0;
+  const Two = (numOfTwoStars / totalNumberOfRatings) * 100;
+
+
+  const numOfOneStars = Number(ratings['1']) || 0;
+  const One = (numOfOneStars / totalNumberOfRatings) * 100;
```
Comment (mictro97) [inline client/src/components/Reviews/RatingBars.jsx:23]: These lines could be made into a helper function to avoid repeating this calculation

File: client/src/components/Reviews/RatingBreakdown.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    for (var key in metaData.ratings) {
+      totalNumberOfRatings += Number(metaData.ratings[key]);
+      sumOfAllRatings += (Number(key) * Number(metaData.ratings[key]));
+    }
+
+    const avgRating = (sumOfAllRatings / totalNumberOfRatings);
+    roundedAvgRatingNumber = Math.round(avgRating * 10) / 10;
+    roundedAvgRatingString = roundedAvgRatingNumber.toFixed(1); // to render uptil 1 decimal point eg. 3 is displayed as "3.0"
+
+    var percentageOfReviewsThatRecommend = (Math.round(((Number(metaData.recommended['true']) / totalNumberOfRatings) * 100) *10)/10).toFixed(1) || 0; //because of tofixed 30 will be displayed as 30.0
```
Comment (mictro97) [inline client/src/components/Reviews/RatingBreakdown.jsx:22]: Unless I'm reading this wrong, if totalNumberOfRatings is zero this will throw a divide by 0 error, maybe add some logic for that edge case.

**Review by mictro97 — APPROVED** [review]
Overall looks great I would just use 'let' instead of 'var' in for variables that you expect the value to change.

File: client/src/components/Reviews/RatingBars.jsx (hunk 1 above)
Comment (Pallavi25Kishore) [reply client/src/components/Reviews/RatingBars.jsx:23]: sounds good, will change the code accordingly, thanks :)

File: client/src/components/Reviews/RatingBreakdown.jsx (hunk 2 above)
Comment (Pallavi25Kishore) [reply client/src/components/Reviews/RatingBreakdown.jsx:22]: Good catch for the avgRating edge case - will look into it and change code accordingly, thank you :) 
i think percentageOfReviewsThatRecommend should work because of "| 0" in the end of the line - so if we divide by 0 - percentageOfReviewsThatRecommend = undefined || 0 - but ill just add a test case for this edge case as well!
